packages/grpc-js/src/subchannel.ts

```diff
@@ -299,24 +300,22 @@ export class Subchannel {
         };
         connectionOptions.servername = sslTargetNameOverride;
       }
+    }
+    /* In all but the most recent versions of Node, http2.connect does not use
+     * the options when establishing plaintext connections, so we need to
+     * establish that connection explicitly. */
+    connectionOptions.createConnection = (authority, option) => {
       if (proxyConnectionResult.socket) {
-        connectionOptions.socket = proxyConnectionResult.socket;
+        return proxyConnectionResult.socket;
+      } else if ('secureContext' in connectionOptions) {
+        return tls.connect(this.subchannelAddress);
```

Comment:
This isn't better because you're not passing in the options. When connecting with TLS and without a proxy, it should simply not be passing a `createConnection` option at all.

Reply:
My typescript is not up to par :( What I really wanted to do in the ssl lock was like this:

```
      if (proxyConnectionResult.socket) {
        connectionOptions.createConnection = (authority, option) => {
          return proxyConnectionResult.socket;
        };
      }
```

But I get typescript errors.

Reply:
Ah, the issue is that TypeScript thinks that `proxyConnectionResult.socket` might become `null` between when you check it and when the callback is called. This should work:

```ts
if (proxyConnectionResult.socket) {
  connectionOptions.createConnection = (authority, option) => {
    return proxyConnectionResult.socket!; //Exclamation mark here
  };
}
```

Reply:
Nice one! updated.